default-enarx-trust-process.md

```diff
@@ -0,0 +1,474 @@
+the **Enarx host agent**.  The **Enarx client agent** also sends an
+error message to the Orchestrator.
+
+If the attestation measurement meets the checking criteria, the **Enarx
+client agent** retrieves or derives the session key from the previous
+communication originating in the **host CPU + CPU firmware** (the exact
+mechanism being CPU architecture-dependent).  At this point, the state
+transition is considered complete, and the TEE instance, loaded with
+the Enarx runtime image, is now an **Empty Keep**, which includes a
+running **Enarx runtime instance**. 
+
+### State 2->3 change
+
+#### State 2 - attested
+
+ - host owner trust domain
+    - host
+    - host CPU + CPU firmware
+    - Enarx host agent
+    - Other workloads
+ - tenant trust domain
+    - Orchestrator
+    - Enarx client agent
+    - **Empty Keep**
+    - tenant workload image
+    - tenant workload image repository
+
+#### State 3 - provisioned
+
+ - host owner trust domain
+    - host
+    - host CPU + CPU firmware
+    - Enarx host agent
+    - other workloads
+ - tenant trust domain
+    - Orchestrator
+    - Enarx client agent
+    - **Provisioned Keep**
+    - tenant workload image
+    - tenant workload image repository
+
+
+#### Changes
+
+The change between state 2 and state 3 is the loading of the **tenant
+Workload Image** into the **Empty Keep**, turning it into a **Provisioned
+Keep**.
+
+During the state 0->1 change transition, the **Orchestrator** provided the
+**Enarx client agent** with one of the following:
+ 1. access to the **tenant workload image repository** and a reference to
+ **tenant workload image** to be provisioned;
+ 2. direct access to the **tenant workload image**, either as a binary in
+ memory or in storage;
+ 3. a reference allowing the **Enarx client agent** to recontact the
+ **Orchestrator** in the event of successful attestation.
+
+The **Enarx client agent** now requires access to the **tenant workload
+image**.
+
+ - In the case of 1. above, the **Enarx client agent** now exercises its
+ access to the **tenant workload image repository** and retrieves the
+ **tenant workload image**.
+ - In the case of 2. above, the **Enarx client agent** already has access
+ to the **tenant workload image**.
+ - In the case of 3. above, the **Enarx client agent** contacts the
+ **Orchestrator**, providing attestation measurement check status
+ information, and receives the **tenant workload image** from the
+ **Orchestrator**.  Note that explicit approval may be included in the
+ communication to allow the **Orchestrator** to check that the message is
+ a "success" and not a "failure": this is a measure to reduce sensitive
+ information leakage in the event of implementation errors.
+
+The **Enarx client agent** gained access to a session key as part of the
+state 1->2 transition.  The **tenant workload image** must be encrypted
+under this session key to be transmitted to the **Empty Keep**.
```

Comment:
The options (1) and (2) seem to imply the persistence of the workload images, while they are encrypted using a session key, which I read as non-persistent. It might be a little clearer if the object lifetimes (images and keys) are stated explicitly.

Reply:
Good point.  I think I've addressed this with new text.